source/common/upstream/upstream_impl.cc

```diff
@@ -1066,17 +1068,19 @@ void PriorityStateManager::updateClusterPrioritySet(
   auto per_locality_shared =
       std::make_shared<HostsPerLocalityImpl>(std::move(per_locality), non_empty_local_locality);
 
-  // If a batch update callback was provided, use that. Otherwise directly update
-  // the PrioritySet.
-  if (update_cb_ != nullptr) {
-    update_cb_->updateHosts(priority, HostSetImpl::partitionHosts(hosts, per_locality_shared),
-                            std::move(locality_weights), hosts_added.value_or(*hosts),
-                            hosts_removed.value_or<HostVector>({}), overprovisioning_factor);
-  } else {
-    parent_.prioritySet().updateHosts(
-        priority, HostSetImpl::partitionHosts(hosts, per_locality_shared),
-        std::move(locality_weights), hosts_added.value_or(*hosts),
-        hosts_removed.value_or<HostVector>({}), overprovisioning_factor);
+  if (hosts) {
```

Comment:
Hmm, if `current_hosts` is `nullptr`, won't we lose any update here that is adding new hosts? Or is it not possible? Maybe some comments on when `current_hosts` can/can't be null, what `hosts_added` and `hosts_removed` can be in that sitution and why it's safe to then elide the update would be good here.

Reply:
Ok. I think the original intent was to initialize the host vector to be empty instead of `nullptr`. I changed the code to set the pointer if necessary -- this seems a safer approach.